src/main/java/ejercicio/banco/repository/PaymentRepository.java

```diff
@@ -2,14 +2,15 @@
 
 import ejercicio.banco.dto.Payment;
 
+import java.sql.SQLException;
 import java.util.List;
 
 public interface PaymentRepository {
-    List<Payment> findAll(String filename);
+    List<Payment> findAll(String filename) throws SQLException;
```

Comment:
This is not a good idea. `SQLException`is a checked (non-runtime exception) and, therefore, it must be handled. Basically, since this is a database exception, it must be handled in the repository layer and not any upper level. The `try-catch`you added in the service must be added in the repository.

Take a look to the previous article (https://www.baeldung.com/java-exceptions) and see the sections 3.1 and 3.2 (checked vs unchecked exceptions).

Reply:
Done on last commit